### PR TITLE
Fix `RepoCard` test on Windows

### PR DESCRIPTION
**Pull request by hanouticelina**
This should fix the test introduced in #2771 on Windows Python 3.8.

### CODE REVIEW

File: tests/test_repocard.py [hunk 1]
```diff
@@ -265,20 +265,25 @@ def test_metadata_eval_result(self):
 
 
 @with_production_testing
-def test_load_from_hub_if_repo_id_or_path_is_a_dir(monkeypatch):
+def test_load_from_hub_if_repo_id_or_path_is_a_dir(monkeypatch, tmp_path):
```
Comment (Wauplin) [inline tests/test_repocard.py:268]: Much better this way

File: tests/test_repocard.py [hunk 2]
```diff
@@ -265,20 +265,25 @@ def test_metadata_eval_result(self):
...
     """
-    with SoftTemporaryDirectory() as tmpdir:
-        monkeypatch.chdir(tmpdir)
-        repo_id = "openai-community/gpt2"
-        (Path(tmpdir) / "openai-community" / "gpt2").mkdir(parents=True)
+    monkeypatch.chdir(tmp_path)
 
-        card = RepoCard.load(repo_id)
-        assert "GPT-2" in str(card)  # loaded from Hub
+    test_dir = tmp_path / "test_dir"
```
Comment (Wauplin) [inline tests/test_repocard.py:277]:
```suggestion
    test_dir = tmp_path / "openai-community"
```

shouldn't it be like this in order to test the edge case ?

Comment (hanouticelina) [reply tests/test_repocard.py]: yes, addressed in [798d94b](https://github.com/huggingface/huggingface_hub/pull/2774/commits/798d94b47a0fd4a0e27b2cad995ff0bc1f1517a4)

**Review by Wauplin — APPROVED** [review]
